.github/workflows/ci-cd.yml

```diff
@@ -311,3 +311,69 @@ jobs:
         run: |
           gh release upload --repo vacanza/holidays ${{ github.ref_name }} dist/*
           gh release upload --repo vacanza/holidays ${{ github.ref_name }} holidays-*-sbom.json
+
+  upload-to-zenodo:
+    name: Publish release to Zenodo
+    if: github.event_name == 'release' && github.event.action == 'published'
+    runs-on: ubuntu-latest
+    needs: [update-github-release]
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download SBOM
+        uses: actions/download-artifact@v4
+        with:
+          name: sbom
+
+      - name: Download signed artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: signed-artifacts
+          path: dist
+
+      - name: Prepare release archive
+        run: |
+          zip -r vacanza-holidays-${{ github.ref_name }}.zip dist/* holidays-*-sbom.json
+
+      - name: Create new Zenodo deposition
+        id: create_deposition
+        run: |
+          response=$(curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.ZENODO_TOKEN }}" \
+            -H "Content-Type: application/json" \
+            -d '{"metadata": {"title": "Vacanza Holidays ${{ github.ref_name }}", "upload_type": "software", "description": "Release ${{ github.ref_name }} of Vacanza Holidays"}}' \
+            https://zenodo.org/api/deposit/depositions)
+          if [[ $(echo "$response" | jq -r '.status // "none"') == "error" ]]; then
+            echo "Error creating deposition: $(echo "$response" | jq -r '.message')"
+            exit 1
+          fi
+          deposition_id=$(echo "$response" | jq -r '.id')
+          echo "DEPOSITION_ID=$deposition_id" >> $GITHUB_ENV
+        env:
+          ZENODO_TOKEN: ${{ secrets.ZENODO_TOKEN }}
```

Comment:
As I noted [here](https://github.com/vacanza/holidays/pull/2317#issue-2885504518), I recommend to add a new version to the existing Zenodo record instead of creating a new one.